client/state/selectors/is-site-on-paid-plan.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Internal dependencies
+ */
+import { getCurrentPlan } from 'state/sites/plans/selectors';
+import { PLAN_FREE } from 'lib/plans/constants';
+
+/**
+ * Returns true if site is on a paid plan, false if the site is not,
+ * or null if the site is unknown.
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @return {?Boolean} Whether site is on a paid plan
+ */
+const isSiteOnPaidPlan = ( state, siteId ) => {
+	const currentPlan = getCurrentPlan( state, siteId );
+
+	if ( ! currentPlan ) {
+		return null;
```

Comment:
I agree. I had copied the logic from `isSiteOnFreePlan`. I think the idea is that we return `null` if we're just not sure whether we're on a free/paid plan or not. But I would agree that we should only return true if we can confirm the nature of the plan. I'll make the change.